runtime/js/40_read_file.js

```diff
@@ -4,13 +4,13 @@
 ((window) => {
   const core = window.Deno.core;
   const { open, openSync } = window.__bootstrap.files;
-  const { readAllInner, readAllSync } = window.__bootstrap.io;
+  const { readAllSyncSized, readAllInnerSized } = window.__bootstrap.io;
 
   function readFileSync(path) {
     const file = openSync(path);
     try {
-      const contents = readAllSync(file);
-      return contents;
+      const { size } = file.statSync();
+      return readAllSyncSized(file, size);
```

Comment:
Won't this introduce an edge case if file size changes (something is written to it) once you started reading it?

Reply:
Yes, I mentioned that in PR's notes:
> This operates under the assumption that files won't change (truncated or extended) whilst being read, which isn't guaranteed but seems fair IMO

Reply:
You're asking for racy/undefined-behaviour if you're doing that, I can't think of any legitimate use case for it.

I think conceptually from the end-user's perspective `readFile`/`readFileSync` are atomic

Reply:
Files being truncated should already be handled, files being extended could be handled, either by allocating an extra byte in the buffer and seeing if that is set or by doing an extra read.

There's really no sane test for that behaviour IMO, such a test would be implicitly coupled to tokio's 16kb read per poll

Reply:
Resolved in last commit: 852062e19f7da8dadac8fc7e2248e8bcec01fcca